### PR TITLE
Wbp collect

### PR DESCRIPTION
**Pull request by joakim-hove**
I have written some MPI stuff:

image

Thing goes as follows:

1. There is a family of summary keywords `WBP, WBP4, WBP5` and `WBP9`. These are different ways to calculate an average well pressure based on the completed cells and their neighbours. 
2. In opm-common there is a class [`PAvgCalculatorCollection`](https://github.com/OPM/opm-common/blob/master/opm/parser/eclipse/EclipseState/Schedule/Well/PAvgCalculatorCollection.hpp) which holds on to the wells which want to calculate the `WBPx` keywords. The most important method for the current PR is the `::index_list()` which returns all the global indices where we need the pressure.
3. This PR will distribute the `index_list` to all PE's and then assemble the requested pressures and update `PAvgCalculatorCollection` instance and pass it to the summary writer.

The actual calculation of WBP in opm-common is not yet complete; but that should be possible to complete only in opm-common with no coupling to opm-simulators.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this opm-common=2146 please

**Comment by joakim-hove** [comment]
jenkins build this opm-common=2146 please

**Comment by joakim-hove** [comment]
Cool if someone in the know when IT comes to mpi Will tale look at this: @blattms, @akva2, @atgeirr

File: ebos/ecloutputblackoilmodule.hh [hunk 1]
```diff
@@ -837,6 +842,8 @@ public:
             if (gasConnectionSaturations_.count(cartesianIdx) > 0) {
                 gasConnectionSaturations_[cartesianIdx] = Opm::getValue(fs.saturation(gasPhaseIdx));
             }
+            if (this->wbpData_.count(cartesianIdx) > 0)
+                this->wbpData_[cartesianIdx] = Opm::getValue(fs.pressure(oilPhaseIdx));
```
Comment (blattms) [inline ebos/ecloutputblackoilmodule.hh:846]: using map.find would save one lookup.

**Review by blattms — COMMENTED** [review]
As far as  I can jusge this seems sound. Not sure how much the calculator has to calculate. If it is not negligible, we might not want to do all of it on one processes.

**Comment by joakim-hove** [comment]
Thanks for looking into this.

> Not sure how much the calculator has to calculate. If it is not negligible, we might not want to do all of it on one processes.

That is really neglible; essentially a sum over the total number of connected cells.

File: ebos/ecloutputblackoilmodule.hh (hunk 1 above)
Comment (joakim-hove) [reply ebos/ecloutputblackoilmodule.hh:846]: I agree - however here I followed the pattern in the surrounding code.

**Comment by joakim-hove** [comment]
jenkins build this please

File: ebos/eclwriter.hh [hunk 2]
```diff
@@ -191,22 +191,37 @@ public:
...
+            wbp_index_list = wbp_calculators.index_list();
         }
+        if (collectToIORank_.isParallel()) {
+#ifdef HAVE_MPI
+            unsigned long size = wbp_index_list.size();
+            MPI_Bcast(&size, 1, MPI_UNSIGNED_LONG, collectToIORank_.ioRank, MPI_COMM_WORLD);
+            if (!collectToIORank_.isIORank())
+                wbp_index_list.resize( size );
 
+            MPI_Bcast(wbp_index_list.data(), size * sizeof(std::size_t), MPI_CHAR, collectToIORank_.ioRank, MPI_COMM_WORLD);
```
[the remark this answers is not in the document]
Comment (joakim-hove) [reply ebos/eclwriter.hh:213]: Thank I have updated.

**Comment by akva2** [comment]
no update pushed.

**Comment by joakim-hove** [comment]
> no update pushed.

Thanks - now it is.

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**